packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -0,0 +1,240 @@
+        <UniqueIDProvider mockOnFirstRender={true} scope="switch">
+            {(ids) => {
+                const uniqueId = id || ids.get("switch");
+
+                return (
+                    <View
+                        onClick={handleClick}
+                        style={[
+                            sharedStyles.switch,
+                            stateStyles.switch,
+                            disabled && sharedStyles.disabled,
+                        ]}
+                        testId={testId}
+                    >
+                        <StyledInput
+                            aria-checked={checked}
+                            aria-describedby={ariaDescribedBy}
+                            aria-label={ariaLabel}
+                            aria-labelledby={ariaLabelledBy}
+                            disabled={disabled}
+                            id={uniqueId}
+                            ref={ref}
+                            role="switch"
+                            // Input is visually hidden because we use a view and span to render
+                            // the actual switch. The input is used for accessibility.
+                            style={sharedStyles.hidden}
+                            type="checkbox"
+                        />
+                        {icon && styledIcon}
+                        <StyledSpan
+                            style={[sharedStyles.slider, stateStyles.slider]}
+                        />
+                    </View>
+                );
+            }}
+        </UniqueIDProvider>
+    );
+});
+
+const sharedStyles = StyleSheet.create({
+    hidden: {
+        opacity: 0,
+        height: 0,
+        width: 0,
+    },
+    switch: {
+        display: "inline-flex",
+        height: "24px",
+        width: "40px",
+        borderRadius: Spacing.small_12,
+        flexShrink: 0,
+        cursor: "pointer",
+        ":hover": {
+            outlineOffset: "1px",
+        },
+        transition: "background-color 0.15s ease-in-out",
+    },
+    disabled: {
+        cursor: "auto",
+        ":hover": {
+            outline: "none",
+        },
+    },
+    slider: {
+        position: "absolute",
+        top: Spacing.xxxxSmall_2,
+        left: Spacing.xxxxSmall_2,
+        height: "20px",
+        width: "20px",
+        borderRadius: "50%",
+        backgroundColor: Color.white,
+        transition: "transform 0.15s ease-in-out",
+    },
+    icon: {
+        position: "absolute",
+        top: Spacing.xxxSmall_4,
+        left: Spacing.xxxSmall_4,
+        zIndex: 1,
+        transition: "0.15s ease-in-out",
+        transitionProperty: "transform, color",
+    },
+});
+
+const styles: Record<string, any> = {};
+const _generateStyles = (
+    checked: boolean,
+    disabled: boolean,
+    clickable: boolean,
+) => {
+    const checkedStyle = `${checked}-${disabled}-${clickable}`;
+    if (sharedStyles[checkedStyle]) {
+        return sharedStyles[checkedStyle];
+    }
```

Comment:
**suggestion:** What do you think about adding a comment here mentioning that the return value in this function is cached to avoid regenerating a `StyleSheet.create` Aphrodite for every re-render?